### PR TITLE
Added the OS parameterization script

### PR DESCRIPTION
**Pull request by mik-dass**
Signed off : mik-dass
Email : mrinald7@gmail.com

### CODE REVIEW

**Review by anmolbabu — APPROVED** [review]
LGTM

**Review by mitchharpur — APPROVED** [review]
Looks good. I assume its tested and working

**Comment by mik-dass** [comment]
@mitchharpur It's verified for the the hypothesis-core-image. The other api-images will require some changes so that they can accept the passed OS parameter. I have a PR ready for the data-api-image. Once it's verified, we will merge this.

File: Dockerfile [hunk 1]
```diff
@@ -9,10 +13,13 @@ ENV NODE_VERSION ${NODE_VERSION:-9.3.0}
...
 
-# Set dependencies
-RUN yum clean all && yum -y install wget
+# Add the os specific default arguments
+ARG OS_IMG_NAME=centos
+ARG OS_IMG_VERSION=7
+
+# Add the os_wrapper script
+ADD os_wrapper.sh ./os_wrapper.sh
+ADD setup_env ./setup_env
```
Comment (anmolbabu) [inline Dockerfile:22]: Does this work?
Shouldn't it be setup_env.sh ?

File: Dockerfile.app [hunk 2]
```diff
@@ -1,28 +1,27 @@
 ARG NODE_VERSION="8.9.2"
 ARG NPM_VERSION="5.5.0"
-FROM registry.devshift.net/fabric8-hdd/nodejs:${NODE_VERSION}_npm_${NPM_VERSION}
-LABEL maintainer="Anmol Babu <anmolbudugutta@gmail.com>"
-
-# Setup App Dir
-RUN mkdir -p /usr/src/app
-WORKDIR /usr/src/app
+ARG OS_REGISTRY="registry.devshift.net/fabric8-hdd"
```
Comment (anmolbabu) [inline Dockerfile.app]: can we call this just REGISTRY or probably APP_REGISTRY I think this name and the one in https://github.com/fabric8-hdd/fabric8-hypothesis-core-image/pull/16/files#diff-3254677a7917c6c01f55212f86c57fbfR5 need not be the same(although they are most likely to be the same)

File: Dockerfile.tests [hunk 3]
```diff
@@ -1,20 +1,17 @@
 ARG NODE_VERSION=8.9.2
 ARG NPM_VERSION=5.5.0
+ARG OS_REGISTRY="registry.devshift.net/fabric8-hdd"
 
-FROM registry.devshift.net/fabric8-hdd/nodejs:${NODE_VERSION}_npm_${NPM_VERSION}
+FROM ${OS_REGISTRY}/nodejs:${NODE_VERSION}_npm_${NPM_VERSION}
```
Comment (anmolbabu) [inline Dockerfile.tests]: Same as https://github.com/fabric8-hdd/fabric8-hypothesis-core-image/pull/16/files#r166513306

File: cico_utils/setup_utils.sh [hunk 4]
```diff
@@ -1,5 +1,8 @@
 #!/bin/bash -ex
 
+DEVSHIFT_USERNAME="mrinald"
```
Comment (anmolbabu) [inline cico_utils/setup_utils.sh]: Please un-commit these

File: constants.sh [hunk 5]
```diff
@@ -15,6 +15,7 @@ OS=(
 )
 
 DEFAULT_PORT=9090
+DEFAULT_OS_REGISTRY="registry.devshift.net/fabric8-hdd"
```
Comment (anmolbabu) [inline constants.sh]: have it as in https://github.com/fabric8-hdd/fabric8-hypothesis-core-image/pull/16/files#diff-3254677a7917c6c01f55212f86c57fbfR1

**Review by anmolbabu — COMMENTED** [review]
1. Please rebase the PR
2. I feel you might need to add copy commands in cico_utils/setup_env.sh to copy these to the image repo tree as in https://github.com/fabric8-hdd/fabric8-hypothesis-core-image/blob/master/cico_utils/setup_env.sh#L23-L33

Also please verify the PR and mark it verified once verified from any *-api-image repo.

Overall the PR looks good to me. Good job on this PR :+1:

File: Dockerfile.tests [hunk 6]
```diff
@@ -38,11 +35,11 @@ EXPOSE ${APP_PORT}
...
 ARG CACHEBUST=1
 
-#Running the tests
+# Running the tests
 RUN npm test
 
-#Running the infra tests
-ENTRYPOINT ["/bin/bash", "./cico_tests/run_infra_tests.sh"]
+# Running the infra tests
+ENTRYPOINT ["/bin/sh", "./cico_tests/run_infra_tests.sh"]
```
Comment (anmolbabu) [inline Dockerfile.tests]: There's actually a problem around this that all scripts have a bash interpreter indicator line in them if so they don't run as expected(basically the shebang line)
Please verify this works for you ... else change the interpreter line in the necessary scripts
A good reference might be https://stackoverflow.com/questions/10376206/what-is-the-preferred-bash-shebang

File: Dockerfile (hunk 1 above)
Comment (mik-dass) [reply Dockerfile:22]: I am adding the entire folder. setup_env is the folder containing the scripts.

File: cico_utils/setup_utils.sh (hunk 4 above)
Comment (mik-dass) [reply cico_utils/setup_utils.sh]: ok

File: Dockerfile (hunk 1 above)
Comment (anmolbabu) [reply Dockerfile:22]: aah yes I thought this was Dockerfile.app in which case moving this source would have been required.
Yes now the one in line 22 suffices

File: Dockerfile.tests (hunk 6 above)
Comment (mik-dass) [reply Dockerfile.tests]: Should I convert to then : #!/usr/bin/env bash?

Comment (anmolbabu) [reply Dockerfile.tests]: I felt so but we need to verify before doing that

**Review by anmolbabu — APPROVED** [review]

**Pull request merged**